src/chatd.cpp

```diff
@@ -2091,9 +2108,8 @@ Chat::SendingItem* Chat::postMsgToSending(uint8_t opcode, Message* msg)
 bool Chat::sendKeyAndMessage(std::pair<MsgCommand*, KeyCommand*> cmd)
 {
     assert(cmd.first);
-    if (cmd.second)
+    if (cmd.second) // if NEWKEY is required for this NEWMSG...
     {
-        cmd.second->setChatId(mChatId);
```

Comment:
It's not needed here anymore, since the `KeyCommand` is now initialized with the corresponding `chatid`